Create Jekyll pages from CMS exports

gregoryjscott opened the pull request:
This sets up a new Jekyll importer for the CMS exports. I ran `jekyll import fw --file _data/city-news-export-test.xml --dir _posts/test` to produce the Jekyll pages in `_posts/test`. (We probably don't want to leave the generated pages in this PR, I just added them so the results could be reviewed.)

This is using a fork of jekyll-import that is currently here https://github.com/gregoryjscott/jekyll-import/tree/fw. If we decide to go this route then we should move the fork to the cityoffortworth organization.

This will close #11.

Tasks
- [x] Decide what to name the new importer (currently named `fw`)
- [x] Remove the generated files
- [x] Move the jekyll-import to the cityoffortworth organization and update the Gemfile

gregoryjscott commented:
@bechatman take another look at the generated content when you get a chance.

Updates
- Add `date` and `end_date` to front matter
- Include date in filename
- Strip out `h2` and `h4`
- Use [reverse_markdown](https://github.com/xijo/reverse_markdown) to convert body to markdown. I'm not sure why it's adding all the extra line feeds.
- Remove `_posts/` from path of default config

Building now puts all the files under `news/` so I think the category is being set correctly from the default front matter.

gregoryjscott commented on _posts/test/2012-08-10-get-a-great-deal-on-a-rain-barrel.md:
```diff
@@ -0,0 +1,38 @@
+title: Get a great deal on a rain barrel
+excerpt: Rain barrels work by capturing rainwater from rooftops by connecting to the
+  home’s downspouts.
+date: 2012-08-10 15:32:56 UTC
```
I'm not sure this is the format we want but the date formatting was driving me crazy so I left it this way for now.

bechatman replied:
I have been using yyyy-mm-ddThh:mm:ss without a time zone because I thought that was what Jekyll required. It looks like it is handling your dates fine, though.

bechatman commented:
@gregoryjscott I may have just created a merge conflict for you. I used collections for our audit reports, so I had to create a defaults section.

gregoryjscott commented:
I rebased and fixed the conflict.

gregoryjscott commented:
OK @bechatman, this is ready for another look. I ran jekyll-import with `--dir _posts/news` and deleted the _posts/test directory.

Changes:
- Switched to the date format that you suggested
- Leave `div` sections as html

I also did some testing with special characters and everything looks fine to me.

bechatman commented:
Well, for some reason I had a merge conflict when trying to sync to the remote branch, so I hope that my attempt to clear that conflict didn't mess anything up. I basically just deleted things that were listed HEAD and kept (if I remember what it said correctly) RETRIEVE. 

As for the content, there are three issues (see 2012-10-08-community-centers-plan-fall-festivities.md for an example of all)

- The dates are quoted with single quotes. It doesn't seem to be impacting the output, though, so that may not be an issue. 
- the content_id isn't in the front-matter. 
- tables are not rendering correctly. I am not sure if that is a redcarpet option we didn't enable or if something is happening with the formatting, but given that all of our tables have class names, it may be better to treat tables like divs and just include them as HTML. 

Other than that, it looks like the defaults are working, so I think we are clear to close #14.

gregoryjscott commented:
Your merge conflict was caused by my rebase. You would have had to rebase to avoid the merge conflict (and that can only be done from the command line, `git rebase origin/import-fw` in this case).

I'm using Ruby's `to_yaml` to produce the front matter and it will surround the value in quotes if it feels that is necessary. In this case, because the dates contain the `:` character it is adding the quotes. This is fine.

I added `content_id` to the front matter and I left tables as html. I also switched to using https://github.com/cityoffortworth/jekyll-import.

Back to you for review.

bechatman commented:
Looks perfect to me. I guess all that is left to do is remove those generated files and make sure to note how I use the forked version of the importer gem.

gregoryjscott commented:
I cleaned up the commits in this `import-fw` branch to only include what is needed. I also added some instructions for importing data to the contributing documentation. See the [Files changed tab](https://github.com/cityoffortworth/FortWorthTexas.gov/pull/24/files) to review what's now included in this PR. All the code for doing the import is the [custom Jekyll importer](https://github.com/cityoffortworth/jekyll-import/tree/fw).

@bechatman you'll get a massive merge conflict if you try to sync this branch again. If you want to test it again then you should delete your local `import-fw` branch first. I tested the import and the build and it worked fine for me. Its ready to merge.

bechatman commented:
Got it... but I don't know how to delete a local branch and pull down the remote. I just use the Github app and site, not command line, so my git knowledge is fairly limited.

bechatman commented:
Also, do we want to go ahead and switch to jekyll 2.5.3?

gregoryjscott commented:
Maybe, but let's do a separate PR for that.

gregoryjscott commented:
I can't figure out how to delete the local branch and pull using the GitHub app so I temporarily pushed this up to a new branch called `import-wf-cleaned` that you can use for testing if you want. I'll delete it when we're done.

gregoryjscott commented:
We agreed this afternoon to merge this. We can open new issues if we need to make further changes to the import.
